app_build_suite/components.py

```diff
@@ -43,6 +43,9 @@ def run(self) -> None:
         self.run_pre_steps()
         self.run_build_steps()
         self.run_cleanup()
+        if self._failed_build is True:
+            logger.error("Exit 1 due to failed build step.")
+            sys.exit(1)
```

Comment:
@piontec I had to add this because for example failed linting did not made circle ci stop

Reply:
Good catch, I think because build errors are handled in a general way in `catch` and we can't exit because we still need to `run_cleanup`, that was never singalled the right way, not only for circleci.